### PR TITLE
3DS & Wii maintanance

### PR DESCRIPTION
**Pull request by Ghabry**
zlib in general updates to 1.2.11 (old URL is 404)
wildmidi to 0.3.11 on Wii, added for 3DS
3DS: ctrulib and citro3d are now included in new devkitARM
3DS: ICU updated to 58.1
Wii: +ICU-iconv 🎊 

The Wii stuff needs a player code change

### CODE REVIEW

**Comment by Ghabry** [comment]
When sdl_con_audio_sin_mixer is finally finished & merged I could remove SDL mixer, btw.

**Comment by carstene1ns** [comment]
You could update to wildmidi 0.4 branch btw, does not make much difference for us (actually it does for Wii, but the audio_decoder supports this).
Sorry for the broken non-ccache build, I had a similar patch, but did not make a pull request yet (https://github.com/EasyRPG/buildscripts/commit/7e30d32055ff851c4a9595cf20133229824ec8e8).

When this one got merged, I will finally rebase my branch, resolve the `XXX` commits (mostly ICU/3DS related) and then we can update the toolchain finally.

File: 3ds/2_build_toolchain.sh [hunk 1]
```diff
@@ -49,11 +51,21 @@ if [ ! -f .patches-applied ]; then
...
+	# Fix wildmidi linking
+	patch -Np0 < wildmidi.patch
+
 	touch .patches-applied
 fi
 
 function set_build_flags {
-	export CFLAGS="-I$WORKSPACE/include -g0 -O2 -mword-relocations -fomit-frame-pointer -ffast-math -march=armv6k -mtune=mpcore -mfloat-abi=hard"
+        if [ "$ENABLE_CCACHE" ]; then
+                export CC="ccache $TARGET_HOST-gcc"
```
Comment (carstene1ns) [inline 3ds/2_build_toolchain.sh]: stray spaces.

File: 3ds/3_cleanup.sh [hunk 2]
```diff
@@ -3,6 +3,6 @@
 ################################################################
 # Cleanup library build folders and other stuff
 
-rm -rf zlib-*/ freetype-*/ harfbuzz-*/ icu/ icu-native/ libmad-*/ libmodplug-*/ libogg-*/ libpng-*/ libvorbis-*/ pixman-*/ mpg123-*/ libsndfile-*/ speexdsp-*/ sdl-wii/ tremor-lowmem/ .patches-applied
-rm -rf ctrulib/ sf2dlib/ lpp-3ds_libraries/
+rm -rf zlib-*/ freetype-*/ harfbuzz-*/ icu/ icu-native/ libmad-*/ libmodplug-*/ libogg-*/ libpng-*/ libvorbis-*/ pixman-*/ mpg123-*/ libsndfile-*/ speexdsp-*/ sdl-wii/ tremor-lowmem/ wildmidi-*/ .patches-applied
```
Comment (carstene1ns) [inline 3ds/3_cleanup.sh]: `sdl-wii` should be removed here (but since `mad`, `modplug` and `ogg` are gone soon, this does not really matter).

**Pull request merged**